ui/raidboss/data/05-shb/raid/e12n.js

```diff
@@ -50,26 +127,109 @@ export default {
+        data.stacks.push(matches.target);
+      },
+      alertText: (data, matches, output) => {
+        if (data.me === matches.target)
+          return output.stackOnYou();
+      },
+      infoText: (data, _, output) => {
+        if (data.stacks.length === 1)
+          return;
+        const names = data.stacks.map((x) => data.ShortName(x)).sort();
+        return output.stacks({ players: names.join(', ') });
+      },
+      outputStrings: {
+        stacks: {
+          en: 'Stack (${players})',
+          de: 'Sammeln (${players})',
+          fr: 'Package (${players})',
+          cn: '分摊 (${players})',
+          ko: '모이기 (${players})',
+        },
+        stackOnYou: {
+          en: 'Stack on YOU',
+          de: 'Sammeln auf DIR',
+          ja: '自分にシェア',
+          fr: 'Package sur VOUS',
+          ko: '나에게 모이기',
+          cn: '集合点名',
+        },
+      },
+    },
+    {
+      id: 'E12N Rapturous Reach Cleanup',
+      netRegex: NetRegexes.headMarker({ id: '003E', capture: false }),
+      delaySeconds: 10,
+      run: (data) => delete data.stacks,
+    },
+    {
+      id: 'E12N Rapturous Reach Single',
       netRegex: NetRegexes.headMarker({ id: '003E' }),
+      condition: (data) => data.seenIntermission,
       response: Responses.stackMarkerOn(),
     },
     {
-      id: 'E12N Diamond Dust Spread',
+      id: 'E12N Diamond Dust Mitigate',
       netRegex: NetRegexes.startsUsing({ id: '5864', source: 'Eden\'s Promise', capture: false }),
-      response: Responses.spread(),
+      condition: Conditions.caresAboutAOE(),
+      response: Responses.aoe(),
     },
     {
       id: 'E12N Diamond Dust Stop',
       netRegex: NetRegexes.startsUsing({ id: '5864', source: 'Eden\'s Promise', capture: false }),
       delaySeconds: 1, // Avoiding collision with the spread call
-      response: Responses.stopMoving('info'),
+      response: Responses.stopMoving('alert'),
     },
     {
       id: 'E12N Frigid Stone',
       netRegex: NetRegexes.headMarker({ id: '0060' }),
       condition: Conditions.targetIsYou(),
       response: Responses.spread(),
     },
+    {
+      id: 'E12N Tether Collect',
+      netRegex: NetRegexes.tether({ id: tetherIds }),
+      run: (data, matches) => {
+        data.tethers = data.tethers || [];
+        data.tethers.push(matches.id);
+      },
+    },
+    {
+      id: 'E12N Cast Release',
+      netRegex: NetRegexes.startsUsing({ id: ['4E2C', '585B', '5861'], capture: false }),
+      preRun: (data) => data.tethers = data.tethers.sort(),
+      delaySeconds: 0.5, // Tethers should be first in the log, but let's be SURE
+      alertText: (data, _, output) => {
+        if (data.tethers.length !== 2)
+          return;
+        // Leviathan's mechanics aren't easily described in a single word,
+        // so we special-case them.
+        const comboStr = data.tethers[0] + data.tethers[1];
+        if (comboStr in primalOutputStrings)
+          return { alertText: output[comboStr]() };
```

Comment:
```suggestion
          return output[comboStr]();
```

Sorry, the file that I sent you had this as a response and not an alertText.